### ISSUE TITLE
pgbenchsetup not working on master/2.X branch...

### ISSUE THREAD

**Issue by sean-** [cockroachdb/cockroach#25225]
Using `master` (f42d65f1d5a5b316ab7e83f1cbca0ddfd7b07bfd), it isn't possible to build `pgbenchsetup` atm.

```
 % go install github.com/cockroachdb/cockroach/pkg/bench/cmd/pgbenchsetup
# github.com/cockroachdb/cockroach/pkg/sql/lex
/home/seanc/go/src/github.com/cockroachdb/cockroach/pkg/sql/lex/predicates.go:60:14: undefined: reservedKeywords
```

This is a drive-by FYI.  The 1.1 branch version of the tool still works.

If there's a better way to establish a baseline single-node performance for crdb than `pgbench`, pointers would be welcome.

**Comment by dt** [comment]
That looks like you just need to build (cockroach) with `make` once, to run the code-generation steps for all the modules. Does that help?

**Comment by dt** [comment]
I think we intentionally gitignored those files since checking in the generated files was creating significant merge conflicts -- it essentially required every change to those packages be completely serialized. I think we decided that losing go-get-ability was sadly the price we had to pay for increased development productivity, though I'll let @benesch and @knz chime in there.

That said, we've started working on a general `workload` tool that has several pre-defined benchmarks it can run, and we'll be distributing per-built binaries of that for supported platforms at some point, so we should probably move `pgbench` into a `workload` longer-term.

**Comment by benesch** [comment]
> I think we intentionally gitignored those files since checking in the generated files was creating significant merge conflicts -- it essentially required every change to those packages be completely serialized. I think we decided that losing go-get-ability was sadly the price we had to pay for increased development productivity, though I'll let @benesch and @knz chime in there.

Yep, exactly. We weighed the tradeoffs here and decided it was more important to optimize for our contributor productivity than to preserve compatibility with `go get`. 

You can avoid the dependency on Node and Yarn by running `make buildshort` instead of `make build`. In your case I think `make pkg/sql/parser/sql.go` would have been sufficient, but of course it's impossible to figure that out without quite a bit of knowledge about our build system.

Note that we do publish a separate repository with everything pre-generated: https://github.com/cockroachdb/cockroach-gen Unfortunately it's not `go get`-compatible because the import paths don't line up. But it works well if you want to depend on Cockroach in another project using dep; if you point the `source` for `github.com/cockroachdb/cockroach` to `github.com/cockroachdb/cockroach-gen`, dep will perform the rewriting for you. 

In short: I'm afraid we've done all we can that doesn't require us to check in generated code. If the Go build system allowed us to run arbitrary code at build-time—like every other build system that comes to mind—we wouldn't be in this bind, but the Go team is steadfast in their decision.
